### PR TITLE
Add documentation for Logger with Transport Client

### LINKED ISSUE #21331: Add documentation on how to use the transport client in 5.0.0+
From multiple topics on the Discourse forum we see users running into issues when using the 5.0 transport client. One issue is that they don't know what Maven coordinates to use to get running. Another issue is that the log4j dependency needs to be exact at the moment, it should be 2.6.2 and not 2.7  [1]. We should document this as part of the [transport client documentation](https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/transport-client.html).

[1] https://discuss.elastic.co/t/issue-with-elastic-search-5-0-0-java-lang-classnotfoundexception-org-apache-logging-log4j-core-async-daemonthreadfactory/64345/7

### PR DESCRIPTION
**Pull request by dadoonet**
Closes #21331

### CODE REVIEW

**Comment by dadoonet** [comment]
@jasontedor Could you review this doc please?

File: docs/java-api/index.asciidoc [hunk 1]
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
 </dependency>
 --------------------------------------------------
 
+=== Log4J2 Logger
```
Comment (jasontedor) [inline docs/java-api/index.asciidoc]: Nit: Log4J2 -> Log4j 2

File: docs/java-api/index.asciidoc [hunk 2]
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
...
+
+You need to also include Log4J2 dependencies:
```
Comment (jasontedor) [inline docs/java-api/index.asciidoc]: Nit: Log4J2 -> Log4j 2

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks @dadoonet. I left some important comments about the log4j version. I also left a nit about the Log4j name. This nit is for consistency with other places in the docs.

File: docs/java-api/index.asciidoc [hunk 3]
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
...
+
+["source","xml",subs="attributes"]
+--------------------------------------------------
+<dependency>
+    <groupId>org.apache.logging.log4j</groupId>
+    <artifactId>log4j-api</artifactId>
+    <version>2.6.2</version>
```
Comment (dadoonet) [inline docs/java-api/index.asciidoc]: Ha right! Thanks a lot. I'll fix that.

**Comment by jasontedor** [comment]
Also, it occurred to me that these versions in this doc could easily go out of date when we upgrade the Log4j version in version.properties. Could you add a commit to the line item there for Log4j to upgrade the version in these docs when a change is made to the version there?

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by dadoonet** [comment]
@jasontedor I updated based on your comments. Thanks!

> Could you add a commit to the line item there for Log4j to upgrade the version in these docs when a change is made to the version there?

I just added that.

Ideally, as I wrote in https://github.com/elastic/docs/issues/4, I wish we could have a real project with all the java code we added in documentation which is running for real.
From this project, using code snippet "a la maven" would be neat.

**Comment by dadoonet** [comment]
Also added 
* 5.0: 6348e54650acba3e0188e38d8905cf037cd722c5 and d91ad4e12f28e1b0e723b6223591464ea5be9b77 (with Log4j 2.6.2 instead of 2.7)
* master (6.0): 5336e72fbb85b4d63c8466f64166b58f26a50bb5

**Pull request merged**